rust/src/lib.rs

```diff
@@ -308,16 +323,34 @@ pub unsafe extern "C" fn zcashlc_create_account(
     db_data_len: usize,
     seed: *const u8,
     seed_len: usize,
+    treestate: *const u8,
+    treestate_len: usize,
+    recover_until: i64,
     network_id: u32,
 ) -> *mut FFIBinaryKey {
+    use zcash_client_backend::data_api::BirthdayError;
+
     let res = catch_panic(|| {
         let network = parse_network(network_id)?;
-        let db_data = unsafe { wallet_db(db_data, db_data_len, network)? };
+        let mut db_data = unsafe { wallet_db(db_data, db_data_len, network)? };
         let seed = Secret::new((unsafe { slice::from_raw_parts(seed, seed_len) }).to_vec());
+        let treestate =
+            TreeState::decode(unsafe { slice::from_raw_parts(treestate, treestate_len) })
+                .map_err(|e| anyhow!("Invalid TreeState: {}", e))?;
+        let recover_until = recover_until.try_into().ok();
+
+        let birthday =
+            AccountBirthday::from_treestate(treestate, recover_until).map_err(|e| match e {
+                BirthdayError::HeightInvalid(e) => {
+                    anyhow!("Invalid TreeState: Invalid height: {}", e)
+                }
+                BirthdayError::Decode(e) => {
+                    anyhow!("Invalid TreeState: Invalid frontier encoding: {}", e)
+                }
```

Comment:
Non-blocking: this should have been part of a `Display` impl on `BirthdayError` (I worked around it here).